test/integration/ethereum/flashMintDex.spec.ts

```diff
@@ -0,0 +1,798 @@
+import "module-alias/register";
+import { Account, Address } from "@utils/types";
+import DeployHelper from "@utils/deploys";
+import { getAccounts, getWaffleExpect } from "@utils/index";
+import { setBlockNumber } from "@utils/test/testingUtils";
+import { ProtocolUtils } from "@utils/common";
+import { ethers } from "hardhat";
+import { utils, BigNumber } from "ethers";
+import {
+  IDebtIssuanceModule,
+  IDebtIssuanceModule__factory,
+  SetToken,
+  SetToken__factory,
+  SetTokenCreator,
+  SetTokenCreator__factory,
+  FlashMintDex,
+  IERC20__factory,
+  IWETH__factory,
+  IBasicIssuanceModule,
+  IBasicIssuanceModule__factory,
+} from "../../../typechain";
+import { PRODUCTION_ADDRESSES } from "./addresses";
+import { ADDRESS_ZERO } from "@utils/constants";
+import { ether } from "@utils/index";
+import { impersonateAccount } from "./utils";
+
+const expect = getWaffleExpect();
+
+enum Exchange {
+  None,
+  Sushiswap,
+  Quickswap,
+  UniV3,
+  Curve,
+}
+
+type SwapData = {
+  path: Address[];
+  fees: number[];
+  pool: Address;
+  exchange: Exchange;
+};
+
+type IssueRedeemParams = {
+  setToken: Address;
+  amountSetToken: BigNumber;
+  componentSwapData: SwapData[];
+  issuanceModule: Address;
+  isDebtIssuance: boolean;
+};
+
+type PaymentInfo = {
+  token: Address;
+  limitAmt: BigNumber;
+  swapDataTokenToWeth: SwapData;
+  swapDataWethToToken: SwapData;
+};
+
+const addresses = PRODUCTION_ADDRESSES;
+
+const swapDataEmpty = {
+  exchange: Exchange.None,
+  fees: [],
+  path: [],
+  pool: ADDRESS_ZERO,
+};
+
+const swapDataUsdcToWeth = {
+  exchange: Exchange.UniV3,
+  fees: [500],
+  path: [addresses.tokens.USDC, addresses.tokens.weth],
+  pool: ADDRESS_ZERO,
+};
+
+const swapDataWethToUsdc = {
+  exchange: Exchange.UniV3,
+  fees: [500],
+  path: [addresses.tokens.weth, addresses.tokens.USDC],
+  pool: ADDRESS_ZERO,
+};
```

Comment:
So this swap data would have to be constructed for any ERC-20 input/output token. And I assume it would be best to find the best path dynamically or could it be hard-coded to paths `[inputToken, WETH]` or `[WETH, outputToken]`? @edkim

Reply:
Correct, we'd need these paths defined for all input/output tokens we let users choose. I was imagining these would be hard-coded. The best priced path can change, but not often.